src/libstd/keyword_docs.rs

```diff
@@ -710,3 +789,79 @@ mod loop_keyword { }
 /// [book]: ../book/ch05-01-defining-structs.html
 /// [reference]: ../reference/items/structs.html
 mod struct_keyword { }
+
+#[doc(keyword = "super")]
+//
+/// The parent of the current [module].
+///
+/// [module]: ../../nightly/book/ch07-02-defining-modules-to-control-scope-and-privacy.html
+mod super_keyword { }
+
+#[doc(keyword = "trait")]
+//
+/// A single interface shared across types.
+mod trait_keyword { }
+
+#[doc(keyword = "true")]
+//
+/// A value of type `bool` representing logical **true**.
+mod true_keyword { }
+
+#[doc(keyword = "type")]
+//
+/// Give a new name to an existing type.
+mod type_keyword { }
+
+#[doc(keyword = "unsafe")]
+//
+/// Opt out of statically enforced memory safety.
+mod unsafe_keyword { }
+
+#[doc(keyword = "use")]
+//
+/// Import or rename items from other crates or modules.
+mod use_keyword { }
+
+#[doc(keyword = "where")]
+//
+/// Add bounds to a type parameter.
+mod where_keyword { }
+
+#[doc(keyword = "while")]
+//
+/// Loop while a condition remains true.
+mod while_keyword { }
+
+// 2018 Edition keywords
+
+#[unstable(feature = "async_await", issue = "50547")]
+#[doc(keyword = "async")]
+//
+/// Return a [`Future`] instead of blocking the current thread.
+///
+/// [`Future`]: ./future/trait.Future.html
+mod async_keyword { }
+
+#[unstable(feature = "async_await", issue = "50547")]
+#[doc(keyword = "await")]
+//
+/// Suspend execution until the result of a [`Future`] is ready.
+///
+/// [`Future`]: ./future/trait.Future.html
+mod await_keyword { }
+
+#[doc(keyword = "dyn")]
+//
+/// Name the type of a [trait object].
+///
+/// [trait object]: ../book/ch17-02-trait-objects.html
+mod dyn_keyword { }
+
+// FIXME: #[doc(keyword = "try")]
+
+#[doc(keyword = "union")]
+//
+/// An unsafe [Algebraic Data Type][ADT] equivalent to a C-style `union`.
```

Comment:
In what way is a union an ADT?

Reply:
It's not, my mistake. See the comments on `enum`.